File: pattern/prefix_sum/209_Minimum Size Subarray Sum/Solution.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int l=0;
        int minLength= Integer.MAX_VALUE;
        int sum=0;
        for(int r=0;r<nums.length;r++){
            sum+=nums[r];
            while(sum>=target){
                minLength=Math.min(minLength,r-l+1);
                sum-=nums[l];
                l++;
            }
        }
       return (minLength == Integer.MAX_VALUE) ? 0 : minLength;

    }
}

File: pattern/prefix_sum/209_Minimum Size Subarray Sum/Solution.js
/**
 * @param {number} target
 * @param {number[]} nums
 * @return {number}
 */
var minSubArrayLen = function(target, nums) {
    let sum = 0;
    let left = -1;
    let right = 0;
    let res = Number.MAX_SAFE_INTEGER;
    while(right < nums.length){
        sum+=nums[right];
        while(sum >= target){
            res = Math.min(res,right - left);
            sum-=nums[left+1];
            left++;
        }
        right++;
    }
    return res==Number.MAX_SAFE_INTEGER?0:res;
};
